### PR TITLE
Fix bit loss in cie_lightness() when doing division to resolve #15331

### LINKED ISSUE #15331: [Bug] 0.14.33 breaks backlight levels
## Describe the Bug

I have a higher number of backlight levels (20) so it's more obvious, but even a smaller number shows this behavior: increasing/decreasing the backlight levels jumps the light intensity all over the place (e.g. starting from backlight level 0 -> intensity 0, 1 -> intensity 50 (estimated), 2 -> intensity 10, 3 -> intensity 200, 4 -> intensity 50, 5 -> intensity 190).

0.14.32 works normally. If I check out the first commit (b47bdee93c5dc39353477126abd344476c736132 ) after that tag it starts to break.

## System Information

 - Keyboard:  Both KBDfans KBD75 rev2 _and_ Treasure type9s2 show this behavior.
 - Operating system: macOS
 - AVR GCC version: 

 - ARM GCC version: 

 - QMK Firmware version:

 - Any keyboard related software installed? 
   - [ ] AutoHotKey
   - [ ] Karabiner
   - [ ] Other:

## Additional Context

Building using latest qmkfm/qmk_cli Docker image.

### PR DESCRIPTION
**Pull request by karlshea**
## Description

Bit loss during division breaks static backlight levels.

@Duckle29 since you authored the PR that made these changes can you confirm this doesn't break what you were trying to accomplish?

## Types of Changes

- [ ] Core
- [x] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* Fixes #15331

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: quantum/backlight/backlight_avr.c [hunk 1]
```diff
@@ -232,19 +232,19 @@ ISR(TIMERx_OVF_vect) {
...
 // See http://jared.geek.nz/2013/feb/linear-led-pwm
 static uint16_t cie_lightness(uint16_t v) {
-    if (v <= ICRx / 12)  // If the value is less than or equal to ~8% of max
+    if (v <= (unsigned long)ICRx / 12)  // If the value is less than or equal to ~8% of max
     {
         return v / 9;  // Same as dividing by 900%
     } else {
         // In the next two lines values are bit-shifted. This is to avoid loosing decimals in integer math.
-        uint32_t y   = (((uint32_t)v + ICRx / 6) << 5) / (ICRx / 6 + ICRx);  // If above 8%, add ~16% of max, and normalize with (max + ~16% max)
+        uint32_t y   = (((uint32_t)v + (unsigned long)ICRx / 6) << 5) / ((unsigned long)ICRx / 6 + ICRx);  // If above 8%, add ~16% of max, and normalize with (max + ~16% max)
```
Comment (tzarc) [inline quantum/backlight/backlight_avr.c:240]:
```suggestion
        uint32_t y   = (((uint32_t)v + (uint32_t)ICRx / 6) << 5) / ((uint32_t)ICRx / 6 + ICRx);  // If above 8%, add ~16% of max, and normalize with (max + ~16% max)
```

File: quantum/backlight/backlight_avr.c [hunk 2]
```diff
@@ -232,19 +232,19 @@ ISR(TIMERx_OVF_vect) {
 
 // See http://jared.geek.nz/2013/feb/linear-led-pwm
 static uint16_t cie_lightness(uint16_t v) {
-    if (v <= ICRx / 12)  // If the value is less than or equal to ~8% of max
+    if (v <= (unsigned long)ICRx / 12)  // If the value is less than or equal to ~8% of max
```
Comment (tzarc) [inline quantum/backlight/backlight_avr.c:235]:
```suggestion
    if (v <= (uint32_t)ICRx / 12)  // If the value is less than or equal to ~8% of max
```

**Review by tzarc — CHANGES_REQUESTED** [review]

File: quantum/backlight/backlight_avr.c (hunk 2 above)
Comment (zvecr) [inline quantum/backlight/backlight_avr.c:235]:
```suggestion
    if (v <= (uint32_t)ICRx / 12)  // If the value is less than or equal to ~8% of max
```

File: quantum/backlight/backlight_avr.c (hunk 1 above)
Comment (zvecr) [inline quantum/backlight/backlight_avr.c:240]:
```suggestion
        uint32_t y   = (((uint32_t)v + (uint32_t)ICRx / 6) << 5) / ((uint32_t)ICRx / 6 + ICRx);  // If above 8%, add ~16% of max, and normalize with (max + ~16% max)
```

**Review by zvecr — COMMENTED** [review]
Should probably  `uint32_t` like the rest of the function.

**Review by zvecr — APPROVED** [review]

File: quantum/backlight/backlight_avr.c (hunk 1 above)
Comment (karlshea) [reply quantum/backlight/backlight_avr.c]: Already on it :-)

**Review by drashna — APPROVED** [review]

**Comment by tzarc** [comment]
CI says the file needs formatting.

**Pull request merged**